src/Fixer/StringNotation/ExplicitStringVariableFixer.php

```diff
@@ -59,17 +67,54 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
             }
 
             $prevToken = $tokens[$index - 1];
-            if (
-                $prevToken->isGivenKind(T_ENCAPSED_AND_WHITESPACE)
-                || $prevToken->isGivenKind(T_START_HEREDOC)
-                || '"' === $prevToken->getContent()
-            ) {
+            if (!$this->isStringPartToken($prevToken)) {
+                continue;
+            }
+
+            $variableTokens = [
+                $index => $token,
+            ];
+            $firstVariableTokenIndex = $index;
+            $lastVariableTokenIndex = $index;
+
+            $nextIndex = $index + 1;
+            while (!$this->isStringPartToken($tokens[$nextIndex])) {
+                $variableTokens[$nextIndex] = $tokens[$nextIndex];
+                $lastVariableTokenIndex = $nextIndex;
+                ++$nextIndex;
+            }
+
+            if (1 === count($variableTokens)) {
                 $tokens->overrideRange($index, $index, [
                     new Token([T_DOLLAR_OPEN_CURLY_BRACES, '${']),
                     new Token([T_STRING_VARNAME, substr($token->getContent(), 1)]),
                     new Token([CT::T_DOLLAR_CLOSE_CURLY_BRACES, '}']),
                 ]);
+            } else {
+                foreach ($variableTokens as $variablePartIndex => $variablePartToken) {
+                    if ($variablePartToken->isGivenKind(T_NUM_STRING)) {
+                        $tokens[$variablePartIndex] = new Token([T_LNUMBER, $variablePartToken->getContent()]);
```

Comment:
OMG to this 😞 
I have to manually change the type of the token: is there a know map?
Maybe this is not the only one, although test cases cover all the allowed syntax cited in the manual:
https://secure.php.net/manual/en/language.types.string.php#language.types.string.parsing.simple